sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -1092,40 +1093,36 @@ public PagedFlux<ConfigurationSetting> listConfigurationSettingsForSnapshot(Stri
      * <!-- src_embed com.azure.data.appconfiguration.configurationasyncclient.listConfigurationSettingsForSnapshotMaxOverload -->
      * <pre>
      * String snapshotName = &quot;&#123;snapshotName&#125;&quot;;
-     * SettingFields[] fields = new SettingFields[] &#123; SettingFields.KEY &#125;;
-     *
-     * client.listConfigurationSettingsForSnapshot&#40;snapshotName, fields&#41;
+     * SettingSelector selector = new SettingSelector&#40;&#41;.setFields&#40;SettingFields.KEY&#41;;
+     * client.listConfigurationSettingsForSnapshot&#40;snapshotName, selector&#41;
      *     .subscribe&#40;setting -&gt;
      *         System.out.printf&#40;&quot;Key: %s, Value: %s&quot;, setting.getKey&#40;&#41;, setting.getValue&#40;&#41;&#41;&#41;;
      * </pre>
      * <!-- end com.azure.data.appconfiguration.configurationasyncclient.listConfigurationSettingsForSnapshotMaxOverload -->
      *
      * @param snapshotName Optional. A filter used get {@link ConfigurationSetting}s for a snapshot. The value should
      * be the name of the snapshot.
-     * @param fields Optional. The fields to select for the query response. If none are set, the service will return the
-     * ConfigurationSettings with a default set of properties.
+     * @param selector Optional. Selector to filter configuration setting results from the service.
      * @return A Flux of ConfigurationSettings that matches the {@code selector}. If no options were provided, the Flux
      * contains all of the current settings in the service.
      * @throws HttpResponseException If a client or service error occurs, such as a 404, 409, 429 or 500.
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
     public PagedFlux<ConfigurationSetting> listConfigurationSettingsForSnapshot(String snapshotName,
-                                                                                SettingFields[] fields) {
+                                                                                SettingSelector selector) {
```

Comment:
listConfigurationSettingsForSnapshot also lists settings. It takes `acceptDateTime`, `snapshotName`, and future added parameters. Note that `keyFilter` and `labelFilter` can not be used when listing settings in a snapshot.

Reply:
It could be better to not reuse this SettingSelector but either create a new optional bag, `SnapshotSettingSelector` to exclude `keyFilter` and `labelFilter`. Or don't use the optional bag.

Reply:
Seems acceptDateTime is not used in the snapshot. So will only change the array type to List.